pysb/simulator/base.py

```diff
@@ -400,22 +411,22 @@ class SimulationResult(object):
     >>> from pysb.simulator import ScipyOdeSimulator
     >>> import numpy as np
     >>> np.set_printoptions(precision=4)
-    >>> sim = ScipyOdeSimulator(model, tspan=np.linspace(0, 40, 10))
+    >>> sim = ScipyOdeSimulator(model, tspan=np.linspace(0, 40, 10), \
+                                integrator_options={'atol': 1e-20})
     >>> simulation_result = sim.run()
 
     ``simulation_result`` is a :class:`SimulationResult` object. An
     observable can be accessed like so:
 
     >>> print(simulation_result.observables['Bax_c0']) \
         #doctest: +NORMALIZE_WHITESPACE
-    [  1.0000e+00   1.1744e-02   1.3791e-04   1.6196e-06   1.9021e-08
-       2.2344e-10   2.6394e-12   4.8067e-14  -6.2097e-14  -7.5308e-15]
+    [  1.0000e+00   1.1744e-02   1.3791e-04   1.6196e-06   1.9020e-08
+       2.2337e-10   2.6232e-12   3.0806e-14   3.6178e-16   4.2492e-18]
 
     It is also possible to retrieve the value of all observables at a
     particular time point, e.g. the final concentrations:
 
-    >>> print(simulation_result.observables[-1]) #doctest: +ELLIPSIS
-    (-7.5308...e-15, -1.6809...e-13, 1.0000...)
+    >>> obs_last_timepoint = simulation_result.observables[-1]
```

Comment:
Hah, we just discovered this issue on our side too. I think it's specifically numpy 1.12 fixed a bug with set_printoptions(precision=0) but in the process changed how `numpy.void` objects are formatted: trailing zeros after the decimal place are suppressed. So 1.0000 becomes "1." and 1.2300 becomes "1.23". Rows from a recarray are returned as `void` which is why we only have this problem when displaying a row from `observables` or `all`. (Presumably `expressions` would have the same problem). I would suggest we go back to the old way of printing the numbers in the doctest, but format the doctest output to match the numpy 1.12 way. Since our Travis setup will use the latest numpy, CI will pass. Anyone running doctests locally will just need the latest numpy (but I don't think we should require '>=12' in `setup.py`.